es-vue-base/src/lib-components/EsVerificationCode.vue

```diff
@@ -119,89 +144,81 @@ export default {
             }
-            this.checkCodeValidity(this.code);
+
+            this.emitCodeUpdate();
         },
         handleInput(event) {
             const { inputType } = event;
-            let currentActiveElement = event.target;
+            const currentActiveElement = event.target;
+            // Getting the current index handles case where user pastes into an input other than input[0]
+            const curIndex = this.getElementIndex(currentActiveElement);
 
             // Triggered on a normal input that passes handleKeys bubble
             if (!inputType || inputType === 'insertText') {
-                this.checkCodeValidity(this.code);
-
+                this.code[curIndex] = event.target.value;
                 // Move to next input if possible
                 currentActiveElement.nextElementSibling?.focus();
             }
 
-            // If this is a paste and we stored the data parse it
-            if (inputType === 'insertFromPaste' && this.dataFromPaste) {
-                // Loop over each character in our paste
-                this.dataFromPaste.forEach((char) => {
-                    // Getting the current index handles case where user pastes into an input other than input[0]
-                    const curIndex = this.getElementIndex(currentActiveElement);
-                    this.code[curIndex] = char;
-
-                    // If the paste is less than this.charCount; move the focus to the first empty input
-                    // If the paste is equal or larger than this.charCount we move focus to the last input
-                    if (currentActiveElement.nextElementSibling) {
-                        // TODO: Why does chromium require setting the value directly?
-                        currentActiveElement.value = char;
-
-                        currentActiveElement = currentActiveElement.nextElementSibling;
-                        currentActiveElement?.focus();
-                    }
-                });
-            }
+            this.emitCodeUpdate();
         },
-        // Test pasting content
         handlePaste(event) {
-            const currentActiveElement = event.target;
-            const prevIndex = this.getElementIndex(currentActiveElement);
-
-            // TODO: Clearing this is not ideal but pasting over existing data doesn't work
-            this.code = Array(this.charCount);
-
             // Get the paste from clipboard; truncate it if its too long
             this.dataFromPaste = event.clipboardData?.getData('text')
-                .trim().split('').splice(0, this.charCount - prevIndex);
+                .trim().split('').splice(0, this.charCount);
 
-            // Check the data only contains accepted characters
-            const checkPasteValidity = this.checkCodeValidity(this.dataFromPaste);
+            // Reset the code to allow new paste
+            this.code = Array(this.charCount).fill('');
+            this.dataFromPaste.forEach((char, index) => {
```

Comment:
What do you think about generating the code-array by mapping over an array of empty strings instead of setting the data to an array of empty strings, then updating inline?

Given that `code` is "reactive" is that "safer"? Maybe not given it's writing to a v-dom?

```js
this.code = Array(this.charCount).fill('').map((cur, index) => {
    const pasteVal = this.dataFromPaste[index];
    return pasteVal ? pasteVal : cur;
});
```